extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts

```diff
@@ -0,0 +1,295 @@
+        'trackException',
+        'trackTrace',
+        'trackMetric',
+        'trackDependencyData',
+        'throwInternal',
+        'logInternalMessage',
+        'triggerSend',
+        '_sender',
+      ]
+    };
+    return config;
+  }
+
+  public identifier: string = DebugPlugin.identifier;
+
+  constructor() {
+    super();
+    /**
+     * the root element of the logger
+     */
+    let rootElement: HTMLDivElement;
+
+    /**
+     * the logging element
+     */
+    let logger: LoggingElement;
+
+    /**
+     * the style that is only on the webpage when the log is active
+     */
+    let tempStyleEl: HTMLStyleElement;
+
+    /**
+     * the style that will be permanently embedded in the webpage
+     * TODO: manage style conflicts (prepend unique ID to relevant class names?)
+     */
+    let permStyleEl: HTMLStyleElement;
+
+    /**
+     * an object containing the individual debug bin items
+     */
+    let debugBins: {
+      [key: string]: DebugBin;
+    };
+
+    /**
+     * the parent containing all the individual debugBins
+     */
+    let debugBinParent: DebugBinParent;
+
+    /**
+     * the different telemetry functions that will be tracked
+     */
+    let trackers: string[];
+
+    /**
+     * appinsights analytics extension instance. useful methods on __proto__
+     */
+    let analyticsExt: IPlugin;
+
+    /**
+     * ajax dependency extension instance. useful methods on __proto__
+     */
+    let ajaxDependencyExt: IPlugin;
+
+    /**
+     * appinsights properties extension instance. useful methods on __proto__
+     */
+    let propertiesExt: IPlugin;
+
+    /**
+     * appinsights channel extension instance.
+     */
+    let channelPluginExt: IPlugin;
+
+    /**
+     * timestamp used to track number of seconds since webpage was loaded
+     */
+    let startTime: number = +new Date();
+
+    /**
+     * the config for this plugin
+     */
+    let _extensionConfig: IDebugPluginConfig = DebugPlugin.getDefaultConfig();
+
+    dynamicProto(DebugPlugin, this, (_self, base) => {
+      _self.initialize = (config: IDebugPluginConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) => {
+        if (!_self.isInitialized()) {
+          base.initialize(config, core, extensions, pluginChain);
+
+          // 1. grabs each extension involved in instrumentation
+          for (let i = 0; i < extensions.length; i++) {
+            if (extensions[i].identifier === 'ApplicationInsightsAnalytics') {
+              analyticsExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AjaxDependencyPlugin') {
+              ajaxDependencyExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AppInsightsPropertiesPlugin') {
+              propertiesExt = extensions[i];
+            }
+            else if (extensions[i].identifier === 'AppInsightsChannelPlugin') {
+              channelPluginExt = extensions[i];
```

Comment:
We will need to change this as part of the next checkin as this is not always the case.
The extensions used by the internal Microsoft Teams don't all use this channel.
Like wise with the Properties plugin.